src/hooks/useDrafts/useDrafts.tsx

```diff
@@ -39,13 +40,18 @@ export const DraftsProvider: React.FC = ({ children }) => {
     videos: [],
   })
 
-  const fetchDrafts = useCallback(async () => {
-    const currentDrafts = await getDrafts()
-    const videos = currentDrafts.filter((draft): draft is VideoDraft => draft.type === 'video')
-    setDraftsState({
-      videos,
-    })
-  }, [setDraftsState])
+  const fetchDrafts = useCallback(
+    async (channelId?: string) => {
+      const currentDrafts = await getDrafts()
+      const videos = currentDrafts
+        .filter((draft): draft is VideoDraft => draft.type === 'video')
+        .filter((draft) => (channelId ? draft.channelId === channelId : true))
+      setDraftsState({
+        videos,
+      })
+    },
+    [setDraftsState]
+  )
```

Comment:
`draftsState` is global and can be reused between views (and they can specify different args)

⬆️ should you give you a hint why there's a problem here